src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -190,11 +256,43 @@ void BaseCouplingScheme::advance()
 
   PRECICE_ASSERT(_couplingMode != Undefined);
 
+  double relativeDt                 = -1;
+  bool   usesFirstParticipantMethod = (_timeWindowSize == -1);
+
+  if (isImplicitCouplingScheme()) {
+    // store data from writeDataContext in buffer, when advance is called.
+    PRECICE_ASSERT(_computedTimeWindowPart > 0);
+    if (not usesFirstParticipantMethod) {
+      relativeDt = _computedTimeWindowPart / _timeWindowSize;
+      PRECICE_ASSERT(math::smallerEquals(relativeDt, 1.0, 10e-9), relativeDt, _computedTimeWindowPart, _timeWindowSize);
+      PRECICE_ASSERT(relativeDt > 0, relativeDt, _computedTimeWindowPart, _timeWindowSize);
+      if (relativeDt > 1.0) {
+        PRECICE_ASSERT(math::smallerEquals(relativeDt, 1.0, 10e-9), relativeDt);
+        relativeDt = 1.0;
+      }
+      storeTimeStepSendData(relativeDt);
+    } else {
+      // We don't support subcycling here, because this is complicated. Therefore, use same strategy like for explicit coupling and just use a single value at end of window.
+      // Possible solution: Don't scale times to [0,1], but leave them as they are. Then we would also allow times > 1. We then have two options:
+      // 1) scale the times back later when the time window size is known (to still benefit from the simpler handling, if all times are scaled to [0,1]).
+      // 2) generally use times in the interval [0, timeWindowSize]. This makes the implementation probably a bit more complicated, but also more consistent.
```

Comment:
I think I understand the problem here. Not sure what the better solution is, but is would be nice to somehow also cover this case.